client/app/competitor/page.tsx

```diff
@@ -42,6 +42,21 @@ const EditorButtons = () => {
     const { setEditorContent } = useEditorContent();
     const fileUploadRef = useRef<HTMLInputElement>(null);
     const [currQuestion] = useAtom(currQuestionAtom);
+    const [selectedLanguage, setSelectedLanguage] = useAtom(selectedLanguageAtom);
+
+    const languageLookup = Object.fromEntries(
+        (currQuestion?.languages ?? []).map((l) => [l.name, l.syntax])
+    );
+
+    if (!selectedLanguage && currQuestion?.languages?.[0]?.syntax) {
+        setSelectedLanguage(currQuestion.languages[0].syntax);
+    }
+
+    const handleLanguageChange = (name: string) => {
+        const syntax = languageLookup[name];
+        if (syntax) setSelectedLanguage(syntax);
```

Comment:
The selected language should only ever contain the name of the language.  The resolution from name -> syntax should only happen at the editor.  If you _really_ want to have the syntax exist as state, you will need to store _both_ the name and the syntax in state, though I don't think that's really necessary.